core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -111,6 +111,9 @@ class Invoker(
         // if the doc revision is missing, then bypass cache
         WhiskAction.get(entityStore, actionid.id, actionid.rev, fromCache = actionid.rev != DocRevision()) onComplete {
             case Success(action) =>
+                // only Exec instances that are subtypes of CodeExec reach the invoker
+                assume(action.exec.isInstanceOf[CodeExec[_]])
```

Comment:
all these type checks are making me think we should type-parametrize `WhiskAction[ExecType]` or something, and have methods defined as `def foo[T <: CodeExec](a: WhiskAction[T])`